mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -61,37 +62,53 @@
 @Slf4j
 class ExecutorStateManagerImpl implements ExecutorStateManager {
     private final Map<TaskExecutorID, TaskExecutorState> taskExecutorStateMap = new HashMap<>();
-    Cache<String, JobRequirements> pendingJobRequests = CacheBuilder.newBuilder()
+
+    /**
+     * Cache the available executors ready to accept assignments. Note these executors' state are not strongly
+     * synchronized and requires state level check when matching.
+     */
+    private final ConcurrentHashMap<ContainerSkuID, NavigableSet<TaskExecutorHolder>> executorsBySkuID = new ConcurrentHashMap<>();
```

Comment:
It seems like using a ConcurrentHashMap may not be necessary in this case, as it is meant to manage actor state.

Reply:
+1